addons/web/static/tests/views/fields/domain_field_tests.js

```diff
@@ -942,4 +942,82 @@ QUnit.module("Fields", (hooks) => {
         await click(target, ".modal-footer .btn-primary");
         assert.containsOnce(target, ".modal", "the domain is invalid: the dialog is not closed");
     });
+
+    QUnit.test("domain field can be foldable", async function (assert) {
+        serverData.models.partner.records[0].foo = "[]";
+
+        await makeView({
+            type: "form",
+            resModel: "partner",
+            resId: 1,
+            serverData,
+            arch: `
+                <form>
+                    <sheet>
+                        <group>
+                            <field name="foo" widget="domain" options="{'model': 'partner_type', 'foldable': true}" />
+                        </group>
+                    </sheet>
+                </form>`,
+        });
+
+        // As the domain is empty, the "Match all records" span should be visible
+        assert.containsOnce(target, ".o_field_domain .d-print-contents span");
+
+        // Unfold the domain
+        await click(target, ".o_field_domain .align-items-center");
```

Comment:
the selector is not very informative

Reply:
I don't have much more to work with here